examples/spec/lcdproc/LCDd.ini

```diff
@@ -2308,17 +2298,16 @@ description = "Specifies the displaytype.
 type = boolean
 check/type = boolean
 default = no
-description = "'no' if display connected serial, 'yes' if connected parallel.
-I.e. serial by default"
+description = "'no' if display connected serial, 'yes' if connected parallel."
 
 [serialvfd/custom-characters]
 type = unsigned_long
 check/type = unsigned_long
 check/validation = ^([1-9]\d*|0)$
 check/validation/match = LINE
 check/validation/message = Must be any positive number including 0
-default = 0
-description = "Number of Custom-Characters. default is display type dependent"
+default = ;default is display type dependent
```

Comment:
can you calculate the default?

Reply:
Each type has its own value that is more or less arbitrary, so you would need some sort of switch statement or chained if-else statements.

But the current code uses the value -83 as an indicator that the default should be used, so we might just set that as a default for now. Also if the value is above 99 a warning is displayed and the default is used, so the allowed value range is only 0-99 as well as -83.

https://github.com/lcdproc/lcdproc/blob/master/server/drivers/serialVFD_displays.h

Reply:
Yes, default value of -83 makes sense, then.

But ideally we would use the conditional plugin to actually calculate the number.

Reply:
> But ideally we would use the conditional plugin to actually calculate the number.

Not sure whether the syntax is correct, but I think the mapping from types to values is correct:
```
assign/condition/#0 = ( ./ == '0' ) ? '1' : '0'
assign/condition/#1 = ( ./ == '1' ) ? '31' : '0'
assign/condition/#2 = ( ./ == '2' ) ? '16' : '0'
assign/condition/#3 = ( ./ == '3' ) ? '3' : '0'
assign/condition/#4 = ( ./ == '4' ) ? '10' : '0'
assign/condition/#5 = ( ./ == '5' ) ? '3' : '0'
assign/condition/#6 = ( ./ == '6' ) ? '0' : '0'
assign/condition/#7 = ( ./ == '7' ) ? '16' : '0'
assign/condition/#8 = ( ./ == '8' ) ? '0' : '0'
```

Reply:
Unfortunately this did not work. The README.md does not show any array example at all. In the code (conditional.c:953) I am not even sure if this is intended to work like this

Reply:
https://github.com/ElektraInitiative/libelektra/blob/b635bc3f69621d7f23e4ff3e3785fbae2b57504f/src/plugins/conditionals/README.md#L54-L55

I just based my solution on these lines in the README, but the plugin implementation is very confusing and not at all documented, so I don't know whether or not this should work somehow.